### DISCUSSION TITLE
Need assistance with testing new cluster

### DISCUSSION THREAD

**Discussion by davehouser1** [strimzi/strimzi-kafka-operator#4745] [Q&A]
Forgive me if this question is out of scope re: strimzi standpoint of view.
Followed these instructions [here](https://strimzi.io/docs/operators/latest/quickstart.html)

Step 4 of Sending and receiving messages from a topic reads:
_Open a terminal, and start the Kafka console producer with the topic my-topic:_
`bin/kafka-console-producer.sh --broker-list <node-address>:_<node-port>_ --topic my-topic`

I downloaded and deployed the binary, built gradlew on my personal PC (not on any nodes of my cluster)
I then ran the producer and consumer steps on my personal PC. 
They both seem to start up, but If I use a different IP or port (which did not appear when trying to run `kubectl get service my-cluster-kafka-external-bootstrap -n my-kafka-project -o=jsonpath='{.spec.ports[0].nodePort}{"\n"}'` and `kubectl get nodes --output=jsonpath='{range .items[*]}{.status.addresses[?(@.type=="InternalIP")].address}{"\n"}{end}'`) and I see the same output. 

Does not seem right.

Also the step that says "Verify that you see the incoming messages in the consumer console."
how? Do I just start typing in one screen? 

can you clarify how to perform a successful test with the deployment I made?

**Comment by davehouser1** [comment]
nm, it works, albite a little unclear for myself.
I was able to also create messages and check on them. I ran this from my master to check on those messages

`kubectl -n kafka run kafka-consumer -ti --image=strimzi/kafka:latest-kafka-2.4.0 --rm=true --restart=Never -- bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --topic my-topic --from-beginning`

**Reply by scholzj** [reply]
The commends with `kubectl run ...` run inside the Kubernetes cluster in a client pod. The commends from the initial question run locally. They might be able to connect also over other combinations of node address or over the broker node ports. It depends a bit on your Kubernetes cluster. You did not shared more details so it is hard to say why exactly it worked.
